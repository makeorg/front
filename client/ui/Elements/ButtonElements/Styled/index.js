import styled from 'styled-components';
import { intToPx } from 'Shared/helpers/styled';
import {
  Breakpoints,
  DefaultPadding,
} from 'Client/app/assets/vars/Breakpoints';
import {
  BasicColors,
  IconColors,
  MakeThemeColors,
  BackgroundColors,
  TextColors,
} from 'Client/app/assets/vars/Colors';
import { MakeFonts } from 'Client/app/assets/vars/Fonts';

export const ButtonsWrapperStyle = styled.div`
  display: flex;
  flex-flow: column;
  width: 100%;
  justify-content: center;
  align-items: center;
  margin: 10px 0;
  @media (min-width: ${intToPx(Breakpoints.Tablet)}) {
    flex-flow: row;
    margin: 20px 0;
  }
`;

export const UnstyledButtonStyle = styled.button`
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0;
  border: none;
  background-color: transparent;
`;

export const BasicButtonStyle = styled.button`
  display: flex;
  justify-content: center;
  align-items: baseline;
  font-family: ${MakeFonts.TradeGothicBoldCondensed};
  font-size: 14px;
  line-height: 1;
  border: none;
  border-radius: 20px;
  text-transform: uppercase;
  padding: 12px 25px 8px;
  @media (min-width: ${intToPx(Breakpoints.Tablet)}) {
    font-size: 16px;
    padding: 13px 25px 10px;
  }
  @media (min-width: ${intToPx(Breakpoints.LargeDesktop)}) {
    font-size: 18px;
  }
`;

export const RedButtonStyle = styled(BasicButtonStyle)`
  color: ${BasicColors.PureWhite};
  background-color: ${MakeThemeColors.Red};
  svg {
    fill: ${BasicColors.PureWhite};
  }
`;

export const GreyButtonStyle = styled(BasicButtonStyle)`
  color: ${TextColors.MediumGrey};
  background-color: ${BackgroundColors.ExtraLightGrey};
  svg {
    fill: ${MakeThemeColors.Red};
  }
  &:hover,
  &:focus {
    color: ${TextColors.MediumGrey};
  }
`;

export const SmallRedButtonStyle = styled(RedButtonStyle)`
  padding: 5px 15px 2.5px;
  @media (min-width: ${intToPx(Breakpoints.Tablet)}) {
    padding: 10px 20px;
  }
`;

export const SmallGreyButtonStyle = styled(GreyButtonStyle)`
  padding: 5px 15px 2.5px;
  @media (min-width: ${intToPx(Breakpoints.Tablet)}) {
    padding: 10px 20px;
  }
`;

export const ActiveButtonStyle = styled(BasicButtonStyle)`
  color: ${BasicColors.PureWhite};
  background-color: ${MakeThemeColors.Red};
  svg,
  > .tofill {
    fill: ${BasicColors.PureWhite};
  }
  &:disabled {
    color: ${TextColors.MediumGrey};
    background-color: ${BackgroundColors.ExtraLightGrey};
    svg,
    .tofill {
      fill: ${TextColors.MediumGrey};
    }
  }
`;

export const IconWrapperStyle = styled.span`
  display: inline-flex;
  justify-content: flex-start;
  align-content: center;
  font-size: 12px;
  margin-right: 5px;
  svg {
    width: 12px;
    height: 12px;
  }
  @media (min-width: ${intToPx(Breakpoints.LargeDesktop)}) {
    svg {
      width: 14px;
      height: 14px;
    }
  }
`;

export const SocialButtonStyle = styled(ActiveButtonStyle)`
  width: 100%;
  margin: 0 5px;
  padding: 5px 15px;
  color: ${BasicColors.PureWhite};
  svg {
    fill: ${BasicColors.PureWhite};
  }
  @media (min-width: ${intToPx(Breakpoints.Tablet)}) {
    padding: 10px 20px;
  }
`;

const SocialButtonMargin = `
  margin-top: 10px;
  @media (min-width: ${intToPx(Breakpoints.Tablet)}) {
    margin-top: 0;
  }
`;

export const FacebookButtonStyle = styled(SocialButtonStyle)`
  background-color: ${IconColors.Facebook};
`;

export const GoogleButtonStyle = styled(SocialButtonStyle)`
  ${SocialButtonMargin};
  background-color: ${IconColors.Google};
`;
export const EmailButtonStyle = styled(SocialButtonStyle)`
  ${SocialButtonMargin};
  background-color: ${MakeThemeColors.Red};
`;

export const RedLinkButtonStyle = styled(UnstyledButtonStyle)`
  display: inline-block;
  color: ${MakeThemeColors.Red};
  text-decoration: underline;
  margin: 0 5px;
`;

export const ButtonSmallWrapperStyle = styled(ButtonsWrapperStyle)`
  max-width: 410px;
`;

export const CloseButtonStyle = styled(UnstyledButtonStyle)`
  position: absolute;
  top: ${intToPx(DefaultPadding.Mobile)};
  right: ${intToPx(DefaultPadding.Mobile)};
  fill: ${MakeThemeColors.Red};
  z-index: 1;
  font-size: 16px;
  @media (min-width: ${intToPx(Breakpoints.Tablet)}) {
    top: ${intToPx(DefaultPadding.Desktop)};
    right: ${intToPx(DefaultPadding.Desktop)};
  }
`;

export const NavButtonStyle = styled(UnstyledButtonStyle)`
  font-family: ${MakeFonts.TradeGothicBoldCondensed};
  font-size: 14px;
  line-height: 20px;
  color: ${MakeThemeColors.Red};
  text-transform: uppercase;
  @media (min-width: ${intToPx(Breakpoints.LargeDesktop)}) {
    font-size: 16px;
  }
`;
