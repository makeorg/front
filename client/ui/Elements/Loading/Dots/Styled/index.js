import styled, { keyframes } from 'styled-components';
import { intToPx } from 'Shared/helpers/styled';
import { Breakpoints } from 'Client/app/assets/vars/Breakpoints';

const PendingAnimation = keyframes`
0%, 50% {
  transform: initial;
}
20% {
  transform: translateY(-5px);
}
`;

export const LoadingWrapperStyle = styled.div`
  display: flex;
  justify-content: space-between;
  justify-self: center;
  align-self: center;
  max-width: 100px;
  margin: 0 auto;
`;

export const DotStyle = styled.span`
  animation-delay: ${props => props.delay}s;
  animation-duration: ${props => props.duration}s;
  animation-name: ${PendingAnimation};
  animation-iteration-count: infinite;
  animation-timing-function: ease-in-out;
  padding: 0 2px;
  font-size: 16px;
  @media (min-width: ${intToPx(Breakpoints.Desktop)}) {
    font-size: 22px;
  }
`;
