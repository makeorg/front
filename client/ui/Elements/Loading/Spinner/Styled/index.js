/* @flow */

import styled, { keyframes } from 'styled-components';
import { pxToRem } from 'Shared/helpers/styled';
import { BackgroundColors } from 'Client/app/assets/vars/Colors';
import { MiddleColumnStyle } from 'Client/ui/Elements/FlexElements';

export const SpinnerContainerStyle = styled(MiddleColumnStyle)`
  width: 100%;
  height: 100%;
`;

export const SpinnerWrapperStyle = styled.div`
  display: block;
  position: relative;
  width: ${pxToRem('64px')};
  height: ${pxToRem('64px')};
  overflow: hidden;
`;

const Spinning = keyframes`
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
`;

export const SpinnerFirstRingStyle = styled.div`
  box-sizing: border-box;
  display: block;
  position: absolute;
  width: ${pxToRem('51px')};
  height: ${pxToRem('51px')};
  margin: ${pxToRem('6px')};
  border: ${pxToRem('6px')} solid ${BackgroundColors.MediumGrey};
  border-radius: 50%;
  animation: ${Spinning} 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite;
  border-color: ${BackgroundColors.MediumGrey} transparent transparent
    transparent;
`;

export const SpinnerSecondRingStyle = styled(SpinnerFirstRingStyle)`
  animation-delay: -0.45s;
`;

export const SpinnerThirdRingStyle = styled(SpinnerFirstRingStyle)`
  animation-delay: -0.3s;
`;

export const SpinnerFourthRingStyle = styled(SpinnerFirstRingStyle)`
  animation-delay: -0.15s;
`;
