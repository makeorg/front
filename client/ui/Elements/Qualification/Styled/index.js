import styled from 'styled-components';
import { pxToRem } from 'Shared/helpers/styled';
import { Breakpoints } from 'Client/app/assets/vars/Breakpoints';
import { MakeFonts } from 'Client/app/assets/vars/Fonts';
import { BasicColors, ShadowColors } from 'Client/app/assets/vars/Colors';

export const ButtonStyle = styled.button`
  display: flex;
  justify-content: space-between;
  align-items: baseline;
  width: 100%;
  border-width: ${pxToRem('2px')};
  font-size: ${pxToRem('12px')};
  line-height: ${pxToRem('26px')};
  border-style: solid;
  padding: 0 ${pxToRem('10px')};
  border-radius: ${pxToRem('36px')};
  border-color: ${props => props.color};
  @media (min-width: ${pxToRem(Breakpoints.Desktop)}){
    font-size: ${pxToRem('16px')};
    line-height: ${pxToRem('33px')};
    padding: 0 ${pxToRem('15px')};
  }
`;

export const UnqualifyButtonStyle = styled(ButtonStyle)`
  color: ${BasicColors.PureWhite};
  background-color: ${props => props.color};
  box-shadow: 0 1px 1px 0 ${ShadowColors.BlackZeroFiveOpacity};
`;

export const QualifyButtonStyle = styled(ButtonStyle)`
  color: ${props => props.color};
  background-color: ${BasicColors.PureWhite};
  &:focus,
  &:active {
    color: ${BasicColors.PureWhite};
    background-color: ${props => props.color};
  }
`;

export const CounterStyle = styled.span`
  font-family: ${MakeFonts.RobotoBold};
  font-size: ${pxToRem('18px')};
  margin-left: ${pxToRem('10px')};
  @media (min-width: ${pxToRem(Breakpoints.Desktop)}){
    font-size: ${pxToRem('22px')};
  }
`;
