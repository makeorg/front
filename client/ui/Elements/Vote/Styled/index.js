import styled, { keyframes } from 'styled-components';
import { BasicColors } from 'Client/app/assets/vars/Colors';
import { intToPx } from 'Shared/helpers/styled';
import { Breakpoints } from 'Client/app/assets/vars/Breakpoints';

export const ButtonStyle = styled.button`
  position: relative;
  z-index: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 38px;
  height: 38px;
  min-width: 38px;
  min-height: 38px;
  border-width: 2px;
  border-style: solid;
  border-radius: 50%;
  border-color: ${props => props.color};
  overflow: hidden;
  padding: 0;
  @media (min-width: ${intToPx(Breakpoints.LargeMobile)}) {
    width: 43px;
    height: 43px;
    min-width: 43px;
    min-height: 43px;
  }
  @media (min-width: ${intToPx(Breakpoints.LargeDesktop)}) {
    width: 48px;
    height: 48px;
    min-width: 48px;
    min-height: 48px;
  }
`;

export const ButtonIconWrapperStyle = styled.span`
  transform: ${props => props.transform};
  svg {
    position: absolute;
    top: 50%;
    left: 50%;
    z-index: 0;
    transform: translate(-50%, -50%);
    width: 16px;
  }
  @media (min-width: ${intToPx(Breakpoints.LargeMobile)}) {
    svg {
      width: 22px;
    }
  }
  @media (min-width: ${intToPx(Breakpoints.LargeDesktop)}) {
    svg {
      width: 25px;
    }
  }
`;

export const IsVotedButtonStyle = styled(ButtonStyle)`
  color: ${BasicColors.PureWhite};
  background-color: ${props => props.color};
  &.wait {
    cursor: wait;
  }
  .tofill {
    fill: ${BasicColors.PureWhite};
  }
`;

const RotateButton = keyframes`
  0% { transform: rotate(0deg); }
  45% { transform: rotate(-20deg); }
  65% { transform: rotate(20deg); }
  100% { transform: rotate(0deg); }
`;

const InverseRotateButton = keyframes`
  0% { transform: rotate(0deg); }
  45% { transform: rotate(20deg); }
  65% { transform: rotate(-20deg); }
  100% { transform: rotate(0deg); }
`;

export const VoteButtonStyle = styled(ButtonStyle)`
  color: ${props => props.color};
  background-color: ${BasicColors.PureWhite};
  .tofill {
    fill: ${props => props.color};
  }
  box-shadow: 0 0 0 0 ${props => props.color};
  transform: scale(1);
  transition: box-shadow 0.1s ease-in, transform 0.1s ease-in;
  &:hover,
  &:focus {
    box-shadow: 0 0 8px 0px ${props => props.color};
    transform: scale(1.1);
  }
  &.animated {
    box-shadow: 0 0 0 0 ${props => props.color};
    animation: ${RotateButton} 0.5s 1;
    transform: scale(0.9);
  }
  &.animated.disagree {
    animation: ${InverseRotateButton} 0.5s 1;
  }
  &.wait {
    cursor: wait;
  }
`;
