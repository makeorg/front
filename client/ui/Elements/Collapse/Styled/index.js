import styled from 'styled-components';
import { Breakpoints } from 'Client/app/assets/vars/Breakpoints';
import { IntToPx } from 'Shared/helpers/styled';
import { BasicColors, ShadowColors } from 'Client/app/assets/vars/Colors';
import { SeparatorStyle } from 'Client/ui/Elements/Separators';
import { UnstyledButtonStyle } from '../../ButtonElements';

export const CollapseWrapperStyle = styled.div`
  padding: 10px;
  background-color: ${BasicColors.PureWhite};
  box-shadow: 0 1px 1px 0 ${ShadowColors.BlackZeroFiveOpacity};
  margin-top: 10px;
  @media (min-width: ${IntToPx(Breakpoints.Desktop)}) {
    margin-top: 20px;
    padding: 20px;
  }
  &:first-child {
    margin-top: 0;
  }
`;

export const CollapseTriggerStyle = styled(UnstyledButtonStyle)`
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  &:disabled {
    color: ${BasicColors.PureBlack};
    cursor: text;
  }
`;

export const CollapseIconStyle = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  transform: ${props => (props.collapsing ? `rotate(0)` : `rotate(90deg)`)};
`;

export const CollapseContentStyle = styled.div`
  overflow: hidden;
  ${props => (props.collapsing ? `height: 0;` : `height: auto;`)};
  @media (min-width: ${IntToPx(Breakpoints.Desktop)}) {
    ${props => (props.forcedexpand ? `height: auto;` : '')}
  }
`;

export const CollapseSeparatorStyle = styled(SeparatorStyle)`
  margin: 8px 0 16px;
`;
