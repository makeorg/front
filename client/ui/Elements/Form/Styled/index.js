import styled from 'styled-components';
import { pxToRem } from 'Shared/helpers/styled';
import { Breakpoints } from 'Client/app/assets/vars/Breakpoints';
import * as ColorVars from 'Client/app/assets/vars/Colors';
import { UnstyledButtonStyle } from 'Client/ui/Elements/ButtonElements';

export const NoStyleTextInputStyle = styled.input`
  border: none;
  background: transparent;
  background-color: transparent;
  padding: 0 ${pxToRem('10px')};
`;

export const FormWrapperStyle = styled.form`
  display: flex;
  flex-flow: column;
  justify-content: center;
  align-items: center;
`;

export const FormStyle = styled(FormWrapperStyle)`
  width: 100%;
  margin: ${pxToRem('10px')} 0 0;
`;

export const InputErrorMessageStyle = styled.p`
  color: ${ColorVars.BorderColors.ErrorRed};
  margin-bottom: ${pxToRem('10px')};
`;

export const LastFakeInputStyle = styled.div`
  display: flex;
  align-items: center;
  width: 100%;
  padding: 0 ${pxToRem('18px')};
  border-radius: ${pxToRem('30px')};
  border: ${pxToRem('1px')} solid ${ColorVars.BorderColors.LightGrey};
`;

export const ClassicFakeInputStyle = styled(LastFakeInputStyle)`
  margin-bottom: ${pxToRem('15px')};
`;

export const FakeInputGreyStyle = styled(ClassicFakeInputStyle)`
  background: ${ColorVars.BackgroundColors.LightGrey};
  background-color: ${ColorVars.BackgroundColors.LightGrey};
  border-color: ${props => (props.hasError ? ColorVars.BorderColors.ErrorRed : ColorVars.BorderColors.LightGrey)};
`;

export const BasicInputStyle = styled(NoStyleTextInputStyle)`
  width: 100%;
  color: ${ColorVars.TextColors.MediumGrey};
  font-size: ${pxToRem('14px')};
  line-height: ${pxToRem('35px')};
  @media (min-width: ${pxToRem(Breakpoints.Desktop)}){
    font-size: ${pxToRem('16px')};
    line-height: ${pxToRem('40px')};
  }
`;

export const IconLabelStyle = styled.label`
  width: ${pxToRem('30px')};
  color: ${ColorVars.MakeThemeColors.Red};
  font-size: ${pxToRem('16px')};
  @media (min-width: ${pxToRem(Breakpoints.Desktop)}){
    font-size: ${pxToRem('18px')};
  }
`;

export const HidePasswordIconStyle = styled(UnstyledButtonStyle)`
  color: ${ColorVars.TextColors.MediumGrey};
`;

export const InlineParagraphStyle = styled.p`
  display: inline;
  font-size: ${pxToRem('14px')};
  @media (min-width: ${pxToRem(Breakpoints.Desktop)}){
    font-size: ${pxToRem('16px')};
  }
`;

export const ExtraParagraphStyle = styled(InlineParagraphStyle)`
  margin-top: ${pxToRem('15px')};
`;

export const ExtraAltParagraphStyle = styled(InlineParagraphStyle)`
  margin-top: ${pxToRem('10px')};
`;

export const ConditionParagraphStyle = styled(InlineParagraphStyle)`
  color: ${ColorVars.TextColors.MediumGrey};
  font-size: ${pxToRem('12px')};
  margin-bottom: ${pxToRem('15px')};
  @media (min-width: ${pxToRem(Breakpoints.Desktop)}){
    font-size: ${pxToRem('14px')};
  }
`;

export const FormErrorsListStyle = styled.ul`
  font-size: ${pxToRem('12px')};
  margin-top: ${pxToRem('10px')};
  margin-bottom: ${pxToRem('10px')};
  color: ${ColorVars.BorderColors.ErrorRed};
`;

export const FormErrorStyle = styled.li`
  list-style: none;
`;
