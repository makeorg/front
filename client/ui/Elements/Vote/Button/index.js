/* @flow */
import * as React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { ButtonStyle, VoteButtonStyle } from 'Client/ui/Elements/Vote/Styled';

type Props = {
  /** Color property passed to Styled Component */
  color: string,
  /** Vote key's Label */
  label: string,
  /** Vote key's Icon */
  icon: string,
  /** Tabindex for interactive items */
  tabIndex: number,
  /** React Element passed to Styled Component to render correct html tag */
  buttonType?: React.Node,
  /** Rotate property passed to Styled Component */
  rotate?: number,
  /** Id of vote button */
  id?: string,
  /** Method called onMouseLeave to hide Tooltip */
  hideTooltip?: (SyntheticEvent<HTMLButtonElement>) => void,
  /** Method called onMouseEnter to  show Tooltip */
  displayTooltip?: (SyntheticEvent<HTMLButtonElement>) => void,
  /** Method called when vote button is clicked */
  handleVote?: (SyntheticEvent<HTMLButtonElement>, string) => void,
}

/**
 * Renders Vote Button element
 */
export const VoteButtonElement = (props: Props) => {
  const {
    color,
    label,
    icon,
    rotate,
    tabIndex,
    id,
    buttonType,
    handleVote,
    displayTooltip,
    hideTooltip
  } = props;

  return (
    <ButtonStyle
      aria-label={label}
      id={id}
      tabIndex={tabIndex}
      color={color}
      rotate={rotate}
      as={buttonType}
      onClick={handleVote}
      onTouchEnd={handleVote}
      onMouseEnter={displayTooltip}
      onMouseLeave={hideTooltip}
      onFocus={displayTooltip}
      onBlur={hideTooltip}
    >
      <FontAwesomeIcon icon={icon} />
    </ButtonStyle>
  );
};

VoteButtonElement.defaultProps = {
  buttonType: VoteButtonStyle,
  hideTooltip: undefined,
  displayTooltip: undefined,
  handleVote: undefined,
  id: 'vote_button_element',
  rotate: 0
};
