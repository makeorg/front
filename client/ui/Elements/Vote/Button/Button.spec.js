import React from 'react';
import snapshotDiff from 'snapshot-diff';
import renderer from 'react-test-renderer';
import { faThumbsUp } from '@fortawesome/free-regular-svg-icons';
import { UnvoteButtonStyle, VoteButtonStyle } from 'Client/ui/Elements/Vote/Styled';
import { VoteButtonElement } from './index';

describe('VoteButtonElement', () => {
  const defaultProps = {
    color: 'red',
    label: 'foo',
    icon: faThumbsUp
  };

  it('must match the snapshot with default Props', () => {
    const component = renderer.create(
      <VoteButtonElement {...defaultProps} />
    );
    expect(component).toMatchSnapshot();
  });

  it('must return the diff between snapshot with a negative vs a postive Tab Index', () => {
    const NegativeTabIndex = renderer.create(
      <VoteButtonElement {...defaultProps} tabIndex="-1" />
    );
    const PositiveTabIndex = renderer.create(
      <VoteButtonElement {...defaultProps} tabIndex="0" />
    );
    expect(snapshotDiff(NegativeTabIndex, PositiveTabIndex)).toMatchSnapshot();
  });

  it('must return the diff between snapshot when proposal is qualified or not', () => {
    const ButtonIsVoted = renderer.create(
      <VoteButtonElement {...defaultProps} buttonType={UnvoteButtonStyle} />
    );
    const ButtonIsNotVoted = renderer.create(
      <VoteButtonElement {...defaultProps} buttonType={VoteButtonStyle} />
    );
    expect(snapshotDiff(ButtonIsVoted, ButtonIsNotVoted)).toMatchSnapshot();
  });
});
