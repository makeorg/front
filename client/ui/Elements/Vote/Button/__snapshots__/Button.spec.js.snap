// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VoteButtonElement must match the snapshot with default Props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 2.875rem;
  height: 2.875rem;
  border-width: 0.125rem;
  font-size: 1.375rem;
  border-style: solid;
  border-radius: 50%;
  border-color: red;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  overflow: hidden;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 2.875rem;
  height: 2.875rem;
  border-width: 0.125rem;
  font-size: 1.375rem;
  border-style: solid;
  border-radius: 50%;
  border-color: red;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  overflow: hidden;
  color: red;
  background-color: rgb(255,255,255);
}

.c1:focus,
.c1:active {
  color: rgb(255,255,255);
  background-color: red;
}

@media (min-width:47.94rem) {
  .c0 {
    width: 3.75rem;
    height: 3.75rem;
    font-size: 1.875rem;
  }
}

@media (min-width:47.94rem) {
  .c1 {
    width: 3.75rem;
    height: 3.75rem;
    font-size: 1.875rem;
  }
}

<button
  aria-label="foo"
  className="sc-1xooriq-0 c0 c1"
  color="red"
  id="vote_button_element"
  rotate={0}
>
  <svg
    icon="thumbs-up"
    lib="@fortawesome/react-fontawesome"
    prefix="far"
  />
</button>
`;

exports[`VoteButtonElement must return the diff between snapshot when proposal is qualified or not 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -1,8 +1,8 @@
  <button
    aria-label="foo"
-   className="sc-1xooriq-0 sc-1xooriq-0 gzgaZN sc-1xooriq-1 cdNpDN"
+   className="sc-1xooriq-0 sc-1xooriq-0 gzgaZN sc-1xooriq-2 culLKY"
    color="red"
    id="vote_button_element"
    rotate={0}
  >
    <svg
`;

exports[`VoteButtonElement must return the diff between snapshot with a negative vs a postive Tab Index 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -2,11 +2,11 @@
    aria-label="foo"
    className="sc-1xooriq-0 sc-1xooriq-0 gzgaZN sc-1xooriq-2 culLKY"
    color="red"
    id="vote_button_element"
    rotate={0}
-   tabIndex="-1"
+   tabIndex="0"
  >
    <svg
      icon="thumbs-up"
      lib="@fortawesome/react-fontawesome"
      prefix="far"
`;
